api/app/Models/User.php

```diff
@@ -605,11 +605,13 @@ public static function scopeEquity(Builder $query, ?array $equity): Builder
             array_push($equityVars, 'is_visible_minority');
         }
 
-        // 3 fields are booleans, one is a jsonb field, isIndigenous = LEGACY_IS_INDIGENOUS
+        // 3 fields are booleans, one is a jsonb field
         $query->where(function ($query) use ($equityVars) {
             foreach ($equityVars as $index => $equityInstance) {
                 if ($equityInstance === 'is_indigenous') {
-                    $query->orWhereJsonContains('indigenous_communities', IndigenousCommunity::LEGACY_IS_INDIGENOUS->name);
+                    foreach (array_column(IndigenousCommunity::cases(), 'name') as $indigenousOption) {
+                        $query->orWhereJsonContains('indigenous_communities', $indigenousOption);
+                    }
```

Comment:
Can we simplify this and simply check that it's not empty? 
`$query->whereJsonLength('indigenous_communities', '>', 0)

Reply:
cool, this reads as if it is the same logic as the current code but more concise, but i was having trouble testing this so i ran `dd($query->toSql());` before the return statement to see what the actual queries being built were.

## current
```
select * from "users" where (("indigenous_communities")::jsonb @> ? or ("indigenous_communities")::jsonb @> ? or ("indigenous_communities")::jsonb @> ? or ("indigenous_communities")::jsonb @> ? or ("indigenous_communities")::jsonb @> ? or ("indigenous_communities")::jsonb @> ?) and "users"."deleted_at" is null
```

## proposed
```
select * from "users" where (jsonb_array_length(("indigenous_communities")::jsonb) > ?) and "users"."deleted_at" is null
```

@tristan-orourke is the proposed query what you intended? i get an sql error trying to run either of these queries in postico or adminer.

Reply:
That's what I was intending, yes. Not sure where why the error 😕

Reply:
the errors were my fault because i wasn't passing anything in to the question mark operator for the prepared statement (thanks @esizer!). 

### correct example
```
select * from "users" where (("indigenous_communities")::jsonb @> '"STATUS_FIRST_NATIONS"'::jsonb or ("indigenous_communities")::jsonb @> '"NON_STATUS_FIRST_NATIONS"'::jsonb or ("indigenous_communities")::jsonb @> '"INUIT"'::jsonb or ("indigenous_communities")::jsonb @> '"METIS"'::jsonb or ("indigenous_communities")::jsonb @> '"OTHER"'::jsonb or ("indigenous_communities")::jsonb @> '"LEGACY_IS_INDIGENOUS"'::jsonb) and "users"."deleted_at" is null;
```

@tristan-orourke the difference i can see in the logic is that the current one looks for the specific values in the `indigenous_communities` field whereas your proposal would look for any values in the `indigenous_communities` field. it leaves a small possibility that if there was a value like "N/A" or something other than the set enums was forced in, it would fulfill your proposed solution. are you ok with that? if so, i'll go ahead and commit it.

Reply:
I'm okay with this because currently we simply pass in the full set of enum values anyway. Meaning that if we added a value which shouldn't be considered "true", we'd need to amend the code either way. So I prefer checking that it is not empty.